### PR TITLE
Fix Acrobat Reader crash in PDF documents with empty ActualText attributes

### LINKED ISSUE #7034: NVDA PDF Actual Text bug
There is currently a serious bug in NVDA as follows. If you add some Actual Text to a PDF tag and then delete the actual text again, an empty Actual Text attribute will be created in the PDF. This will cause NVDA to stop working completely (remain completely silent) on a Windows 7 machine. On a Windows 8 or Windows 10 machine, Acrobat or Adobe Reader will crash 100% of the time if you try to open such a file with NVDA running. This bug dos not affect JAWS, ZoomText, SuperNova, MAGic, Read&Write Gold or Claro Read.

The attached file that includes the word "Edit" is so affected (will crash). I also attach a version ("Original") which works fine.

For info I have experienced this bug in perhaps 50 different PDFs before I worked out what was going on (not just the attached files).  

Thanks 

[Coram Life Education Bro 2015 - Edit.pdf](https://github.com/nvaccess/nvda/files/883748/Coram.Life.Education.Bro.2015.-.Edit.pdf)
[Coram Life Education Bro 2015 - Original.pdf](https://github.com/nvaccess/nvda/files/883749/Coram.Life.Education.Bro.2015.-.Original.pdf)

### PR DESCRIPTION
**Pull request by jcsteh**
adobeAcrobat vbuf backend: get_PDDomNode can fail and return null, so check for a null pointer before trying to use it.

Fixes #7034. Fixes the first part of #7021.

Changes entry: in Bug Fixes:

```
- Adobe Acrobat Reader no longer crashes in certain PDF documents (specifically, those containing empty ActualText attributes). (#7021, #7034)
```

### CODE REVIEW

File: nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp [hunk 1]
```diff
@@ -417,6 +417,8 @@ AdobeAcrobatVBufStorage_controlFieldNode_t* AdobeAcrobatVBufBackend_t::fillVBuf(
 	}
 
 	IPDDomNode* domNode = getPDDomNode(varChild, servprov);
+	if (!domNode)
+		LOG_DEBUGWARNING(L"Couldn't get IPDDomNode for docHandle " << docHandle << L" id " << ID);
```
Comment (feerrenrut) [inline nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp]: I would suggest wrapping this block in braces.

File: nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp [hunk 2]
```diff
@@ -489,7 +491,7 @@ AdobeAcrobatVBufStorage_controlFieldNode_t* AdobeAcrobatVBufBackend_t::fillVBuf(
 	LOG_DEBUG(L"childCount is "<<childCount);
 
 	bool deletePageNum = false;
-	if (!pageNum && (pageNum = this->getPageNum(domNode)))
+	if (!pageNum && domNode && (pageNum = this->getPageNum(domNode)))
 		deletePageNum = true;
```
Comment (feerrenrut) [inline nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp]: Also braces on this if block

File: nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp [hunk 3]
```diff
@@ -692,11 +694,14 @@ AdobeAcrobatVBufStorage_controlFieldNode_t* AdobeAcrobatVBufBackend_t::fillVBuf(
...
-			// There was text.
-			previousNode = tempNode;
-		}
+		if (domNode) {
+			tempNode = renderText(buffer, parentNode, previousNode, domNode, domElement, useNameAsContent, parentNode->language, textFlags, pageNum);
+			if (tempNode) {
+				// There was text.
+				previousNode = tempNode;
+			}
+		} else
```
Comment (feerrenrut) [inline nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp]: Also braces

File: nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp [hunk 4]
```diff
@@ -692,11 +694,14 @@ AdobeAcrobatVBufStorage_controlFieldNode_t* AdobeAcrobatVBufBackend_t::fillVBuf(
 		}
 
 		// Hereafter, tempNode is the text node (if any).
-		tempNode = renderText(buffer, parentNode, previousNode, domNode, domElement, useNameAsContent, parentNode->language, textFlags, pageNum);
-		if (tempNode) {
-			// There was text.
-			previousNode = tempNode;
-		}
+		if (domNode) {
```
Comment (feerrenrut) [inline nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp]: Should this be checking `pageNum` is not `null_ptr` also?

**Review by feerrenrut — CHANGES_REQUESTED** [review]

File: nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp (hunk 4 above)
Comment (jcsteh) [inline nvdaHelper/vbufBackends/adobeAcrobat/adobeAcrobat.cpp]: No; pageNum can be NULL if there is no page number and `renderText` checks for this.

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**